apps/crud/src/app/app.component.ts

```diff
@@ -1,52 +1,115 @@
-import { CommonModule } from '@angular/common';
-import { HttpClient } from '@angular/common/http';
-import { Component, OnInit } from '@angular/core';
-import { randText } from '@ngneat/falso';
+import {CommonModule} from '@angular/common';
+import {ChangeDetectionStrategy, Component} from '@angular/core';
+import {Todo, TodosService} from './todos.service';
+import {concatMap, map, startWith, Subject} from 'rxjs';
+import {RxState} from '@rx-angular/state';
+import {ForModule, LetModule} from '@rx-angular/template';
+import {MatProgressSpinnerModule} from '@angular/material/progress-spinner';
+import {IfModule} from '@rx-angular/template/experimental/if';
+import {MatSnackBar} from '@angular/material/snack-bar';
+import {createErrorHandler} from './error-handler';
+
+type TodoUi = Todo & {
+  isLoading: boolean;
+};
 
 @Component({
   standalone: true,
-  imports: [CommonModule],
+  imports: [
+    CommonModule,
+    ForModule,
+    MatProgressSpinnerModule,
+    IfModule,
+    LetModule,
+  ],
   selector: 'app-root',
   template: `
-    <div *ngFor="let todo of todos">
-      {{ todo.title }}
-      <button (click)="update(todo)">Update</button>
-    </div>
+    <mat-spinner *rxIf="isLoading$; else todosTemplate"></mat-spinner>
+    <ng-template #todosTemplate>
+      <div *rxFor="let todo of todos$; let index = index">
+        {{ todo.title }}
+        <button (click)="updatesSubject.next(todo)" [disabled]="todo.isLoading">
+          Update
+        </button>
+        <button
+          (click)="deleteSubject.next({todo, index})"
+          [disabled]="todo.isLoading">
+          X
+        </button>
+      </div>
+    </ng-template>
```

Comment:
Depending on what UX you want to offer, you can create a `todoItemComponent` to have a loader per item instead of a global one